src/main/java/eggy/task/TaskList.java

```diff
@@ -73,6 +74,16 @@ public boolean isEmpty() {
         return this.tasks.isEmpty();
     }
 
+    /**
+     * Returns a list of tasks that match the keyword.
+     *
+     * @param keyword The keyword to search for.
+     * @return List of tasks that match the keyword.
+     */
+    public List<Task> findMatchingTasks(String keyword) {
+        return this.tasks.stream().filter(task -> task.getName().contains(keyword)).collect(Collectors.toList());
+    }
+
```

Comment:
I liked how you distinguished between single-valued and multi-valued variables, by using the plural form for the method name since it returns a list of tasks.